streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -77,7 +77,9 @@ public void process(K key, Change<V> change) {
             V newValue = computeValue(key, change.newValue);
             V oldValue = sendOldValues ? computeValue(key, change.oldValue) : null;
 
-            if (sendOldValues && oldValue == null && newValue == null) return; // unnecessary to forward here.
+            if ((sendOldValues && oldValue == null && newValue == null) ||
+                (change.oldValue == null && change.newValue == null)) return; // unnecessary to forward here.
```

Comment:
Actually I think we do not need the second condition here, since a better fix would be in the `mapValues` function, do not forward if both are null, instead of checking them here.

Also minor comment: use a new line for "return" for better debugging, and move the comment on top of line 80 as "if both the new and old values are null after the filtering, do not need to forward downstream anymore".